File: hackerrank/linked-list/dedup.py
# https://www.hackerrank.com/challenges/delete-duplicate-value-nodes-from-a-sorted-linked-list/problem

def RemoveDuplicates(head):
    if head is None:
        return None
    
    curr = head
    while curr.next is not None:
        currentData = curr.data
        next = curr.next;
        nextData = next.data
        
        if currentData == nextData:
            curr.next = curr.next.next
        else:   
            curr = curr.next    
    return head

File: hackerrank/linked-list/dedup.cpp
Node* RemoveDuplicates(Node *head)
{
    if(head == nullptr)
        return nullptr;
    
    if(head->next == nullptr)
        return head;
    
    if(head->data != head->next->data)
    {
        head->next = RemoveDuplicates(head->next);
        return head;
    }

    return RemoveDuplicates(head->next);
}